skills_DS/api/views.py

```diff
@@ -18,11 +19,14 @@ class FileUploadView(APIView):
 	parser_classes = (FileUploadParser,)
 
 	def post(self, request, format=None):
-		file_obj = request.data['file']
+		file_obj = request.FILES['file']
 		# do something with the file
 		if(file_obj):
-			print(file_obj)
+			try:
+				fs = FileSystemStorage()
+				fs.save(request.user.get_full_name().replace(" ", "_") + ".pdf", file_obj)
```

Comment:
I would use `get_username()` instead of `get_full_name()` to obtain a unique filename.

Reply:
If we are using `get_username()` we need to consider the special characters that are allowed in username: `_`, `@`, `+`, `.` and `-`. `-` and `_` shouldn't be a problem, but `@`,  `+`, `.` will need to be replaced by something else.

Reply:
And I think it will be a good idea to generate a uuid in addition to the username to accommodate multiple files from the same user.